### ISSUE TITLE
Windows installation `pip install -e .` error

### ISSUE THREAD

**Issue by 007egleagent** [freqtrade/freqtrade#2529]
I'm running on windows and python version 3.8

I'm running into difficulties downloading the bot on windows.
I got through all of the installation steps for windows here:

```
>cd \path\freqtrade-develop
>python -m venv .env
>.env\Scripts\activate.bat
REM optionally install ta-lib from wheel
REM >pip install TA_Lib‑0.4.17‑cp36‑cp36m‑win32.whl
>pip install -r requirements.txt
>pip install -e .
>freqtrade
```

but i get an error at `pip install -e .`

error log:
```
(.env) C:\Users\rlpiq\freqtrade>pip install -e .
Obtaining file:///C:/Users/rlpiq/freqtrade
Collecting ccxt
Using cached https://files.pythonhosted.org/packages/42/9f/5f2a36ffbced7a2f76d37fc705f7912e3e4176605b66701904f9b76d34d9/ccxt-1.19.44-py2.py3-none-any.whl
Collecting SQLAlchemy
Using cached https://files.pythonhosted.org/packages/34/5c/0e1d7ad0ca52544bb12f9cb8d5cc454af45821c92160ffedd38db0a317f6/SQLAlchemy-1.3.11.tar.gz
Collecting python-telegram-bot
Using cached https://files.pythonhosted.org/packages/26/aa/82dfedea526128a55429527ca5e43c023846e00cc3df30c8dbfebe94bbb2/python_telegram_bot-12.2.0-py2.py3-none-any.whl
Collecting arrow
Using cached https://files.pythonhosted.org/packages/a1/39/a8e116084cf4160f0821ca9bb84ec312ccca84caa2b2bffb70d95d47f91f/arrow-0.15.4-py2.py3-none-any.whl
Collecting requests
Using cached https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl
Collecting urllib3
Using cached https://files.pythonhosted.org/packages/b4/40/a9837291310ee1ccc242ceb6ebfd9eb21539649f193a7c8c86ba15b98539/urllib3-1.25.7-py2.py3-none-any.whl
Collecting wrapt
Using cached https://files.pythonhosted.org/packages/23/84/323c2415280bc4fc880ac5050dddfb3c8062c2552b34c2e512eb4aa68f79/wrapt-1.11.2.tar.gz
Collecting pandas
Using cached https://files.pythonhosted.org/packages/78/b9/a304328ea14cd172a5cf681b634b99e24a5b4e24de83204b713b088f02d5/pandas-0.25.3-cp38-cp38-win32.whl
Collecting scikit-learn
Using cached https://files.pythonhosted.org/packages/1e/ce/9d8c88e68af0a5b5c5d78d8d2b7bcadfd45e1d6afc863ccb9aee30765b06/scikit-learn-0.21.3.tar.gz
Requirement already satisfied: scipy in c:\users\rlpiq\freqtrade\.env\lib\site-packages (from freqtrade==0.17.3) (1.3.2)
Collecting joblib
Using cached https://files.pythonhosted.org/packages/8f/42/155696f85f344c066e17af287359c9786b436b1bf86029bb3411283274f3/joblib-0.14.0-py2.py3-none-any.whl
Collecting jsonschema
Using cached https://files.pythonhosted.org/packages/ce/6c/888d7c3c1fce3974c88a01a6bc553528c99d3586e098eee23e8383dd11c3/jsonschema-3.1.1-py2.py3-none-any.whl
Requirement already satisfied: TA-Lib in c:\users\rlpiq\freqtrade\.env\lib\site-packages (from freqtrade==0.17.3) (0.4.17)
Collecting tabulate
Using cached https://files.pythonhosted.org/packages/66/d4/977fdd5186b7cdbb7c43a7aac7c5e4e0337a84cb802e154616f3cfc84563/tabulate-0.8.5.tar.gz
Collecting cachetools
Using cached https://files.pythonhosted.org/packages/2f/a6/30b0a0bef12283e83e58c1d6e7b5aabc7acfc4110df81a4471655d33e704/cachetools-3.1.1-py2.py3-none-any.whl
Collecting coinmarketcap
Using cached https://files.pythonhosted.org/packages/a8/da/c64662a91905017f237f5ff2778b68638946b0d6268513efadd4d1363669/coinmarketcap-5.0.3.tar.gz
Collecting scikit-optimize
Using cached https://files.pythonhosted.org/packages/f4/44/60f82c97d1caa98752c7da2c1681cab5c7a390a0fdd3a55fac672b321cac/scikit_optimize-0.5.2-py2.py3-none-any.whl
Collecting python-rapidjson
Using cached https://files.pythonhosted.org/packages/42/62/f5a21f199a6c31d52cb97e6c79ebcd77a35ec44a4ccfb10d14c64fdf23d8/python_rapidjson-0.9.1-cp38-cp38-win32.whl
Collecting py_find_1st
Using cached https://files.pythonhosted.org/packages/25/e1/5e274480390df4d2d5b2d46a72ee1abba8de8e883c6f1b32f0c68652880c/py_find_1st-1.1.4.tar.gz
Installing build dependencies ... done
Getting requirements to build wheel ... done
  Preparing wheel metadata ... done
Requirement already satisfied: setuptools>=38.5.1 in c:\users\rlpiq\freqtrade\.env\lib\site-packages (from ccxt->freqtrade==0.17.3) (41.2.0)
Collecting certifi>=2018.1.18
Using cached https://files.pythonhosted.org/packages/18/b0/8146a4f8dd402f60744fa380bc73ca47303cccf8b9190fd16a827281eac2/certifi-2019.9.11-py2.py3-none-any.whl
Collecting yarl==1.1.0; python_version >= "3.5.2"
Using cached https://files.pythonhosted.org/packages/a3/08/05b2d731ef5163b3bcb993f569c4d2b303642f7ee3cbdea373f59e4bd42a/yarl-1.1.0.tar.gz
Collecting aiohttp==3.6.2; python_version >= "3.5.2"
Using cached https://files.pythonhosted.org/packages/c2/f7/f0ad3dbace4762fef5d80aa4124b41bf218e4c4dd6d387a86cede707d9a4/aiohttp-3.6.2-py3-none-any.whl
Collecting aiodns==1.1.1; python_version >= "3.5.2"
Using cached https://files.pythonhosted.org/packages/bd/f5/b69cb930fd5ab0569396659afe3f3c0d37d4098e5d0ba6afdf6fd9388cb0/aiodns-1.1.1-py2.py3-none-any.whl
Collecting cryptography>=2.6.1
Using cached https://files.pythonhosted.org/packages/74/e1/6e360b4dbd2a63c1f506402bfa6fa56aa5826337830d8ee86e5e2ec9457b/cryptography-2.8-cp38-cp38-win32.whl
Collecting future>=0.16.0
Using cached https://files.pythonhosted.org/packages/45/0b/38b06fd9b92dc2b68d58b75f900e97884c45bedd2ff83203d933cf5851c9/future-0.18.2.tar.gz
Collecting tornado>=5.1
Using cached https://files.pythonhosted.org/packages/30/78/2d2823598496127b21423baffaa186b668f73cd91887fcef78b6eade136b/tornado-6.0.3.tar.gz
Collecting python-dateutil
Using cached https://files.pythonhosted.org/packages/d4/70/d60450c3dd48ef87586924207ae8907090de0b306af2bce5d134d78615cb/python_dateutil-2.8.1-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2
Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting idna<2.9,>=2.5
Using cached https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl
Requirement already satisfied: numpy>=1.13.3 in c:\users\rlpiq\freqtrade\.env\lib\site-packages (from pandas->freqtrade==0.17.3) (1.17.4)
Collecting pytz>=2017.2
Using cached https://files.pythonhosted.org/packages/e7/f9/f0b53f88060247251bf481fa6ea62cd0d25bf1b11a87888e53ce5b7c8ad2/pytz-2019.3-py2.py3-none-any.whl
Collecting importlib-metadata
Using cached https://files.pythonhosted.org/packages/f6/d2/40b3fa882147719744e6aa50ac39cf7a22a913cbcba86a0371176c425a3b/importlib_metadata-0.23-py2.py3-none-any.whl
Collecting pyrsistent>=0.14.0
Using cached https://files.pythonhosted.org/packages/30/86/53a88c0a57698fa228db29a4000c28f4124823010388cb7042fe6e2be8dd/pyrsistent-0.15.5.tar.gz
Collecting six>=1.11.0
Using cached https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting attrs>=17.4.0
Using cached https://files.pythonhosted.org/packages/a2/db/4313ab3be961f7a763066401fb77f7748373b6094076ae2bda2806988af6/attrs-19.3.0-py2.py3-none-any.whl
Collecting requests_cache>=0.4.13
Using cached https://files.pythonhosted.org/packages/7f/55/9b1c40eb83c16d8fc79c5f6c2ffade04208b080670fbfc35e0a5effb5a92/requests_cache-0.5.2-py2.py3-none-any.whl
Collecting multidict>=4.0
Downloading https://files.pythonhosted.org/packages/7f/8f/b3c8c5b062309e854ce5b726fc101195fbaa881d306ffa5c2ba19efa3af2/multidict-4.5.2.tar.gz (105kB)
   |████████████████████████████████| 112kB 656kB/s
Collecting async-timeout<4.0,>=3.0
Downloading https://files.pythonhosted.org/packages/e1/1e/5a4441be21b0726c4464f3f23c8b19628372f606755a9d2e46c187e65ec4/async_timeout-3.0.1-py3-none-any.whl
Collecting pycares>=1.0.0
Using cached https://files.pythonhosted.org/packages/85/de/cd46a73e43e206a6ad1e9cf9cc893c3ed1b21caf57f1e0a8d9a119d290eb/pycares-3.0.0.tar.gz
  ERROR: Command errored out with exit status 1:
   command: 'c:\users\rlpiq\freqtrade\.env\scripts\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\rlpiq\\AppData\\Local\\Temp\\pip-install-orj2lfvb\\pycares\\setup.py'"'"'; __file__='"'"'C:\\Users\\rlpiq\\AppData\\Local\\Temp\\pip-install-orj2lfvb\\pycares\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\rlpiq\AppData\Local\Temp\pip-install-orj2lfvb\pycares\pip-egg-info'
       cwd: C:\Users\rlpiq\AppData\Local\Temp\pip-install-orj2lfvb\pycares\
  Complete output (108 lines):
  b'\\\x00I\x00n\x00t\x00e\x00l\x00\\\x00I\x00n\x00t\x00e\x00l\x00(\x00R\x00)\x00 \x00w\x00a\x00s\x00 \x00u\x00n\x00e\x00x\x00p\x00e\x00c\x00t\x00e\x00d\x00 \x00a\x00t\x00 \x00t\x00h\x00i\x00s\x00 \x00t\x00i\x00m\x00e\x00.\x00\r\x00\n\x00'
  Traceback (most recent call last):
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\msvc.py", line 489, in _find_latest_available_vc_ver
      return self.find_available_vc_vers()[-1]
  IndexError: list index out of range

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 154, in save_modules
      yield saved
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 195, in setup_context
      yield
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 250, in run_setup
      _execfile(setup_script, ns)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 45, in _execfile
      exec(code, globals, locals)
    File "C:\Users\rlpiq\AppData\Local\Temp\easy_install-0wy4r5md\cffi-1.13.2\setup.py", line 127, in <module>
    File "C:\Users\rlpiq\AppData\Local\Temp\easy_install-0wy4r5md\cffi-1.13.2\setup.py", line 105, in uses_msvc
    File "C:\Users\rlpiq\AppData\Local\Programs\Python\Python38-32\lib\distutils\command\config.py", line 225, in try_compile
      self._compile(body, headers, include_dirs, lang)
    File "C:\Users\rlpiq\AppData\Local\Programs\Python\Python38-32\lib\distutils\command\config.py", line 132, in _compile
      self.compiler.compile([src], include_dirs=include_dirs)
    File "C:\Users\rlpiq\AppData\Local\Programs\Python\Python38-32\lib\distutils\_msvccompiler.py", line 360, in compile
      self.initialize()
    File "C:\Users\rlpiq\AppData\Local\Programs\Python\Python38-32\lib\distutils\_msvccompiler.py", line 253, in initialize
      vc_env = _get_vc_env(plat_spec)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\msvc.py", line 185, in msvc14_get_vc_env
      return EnvironmentInfo(plat_spec, vc_min_ver=14.0).return_env()
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\msvc.py", line 843, in __init__
      self.si = SystemInfo(self.ri, vc_ver)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\msvc.py", line 485, in __init__
      self.vc_ver = vc_ver or self._find_latest_available_vc_ver()
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\msvc.py", line 492, in _find_latest_available_vc_ver
      raise distutils.errors.DistutilsPlatformError(err)
  distutils.errors.DistutilsPlatformError: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": https://visualstudio.microsoft.com/downloads/

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\rlpiq\AppData\Local\Temp\pip-install-orj2lfvb\pycares\setup.py", line 15, in <module>
      setup(name             = 'pycares',
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\__init__.py", line 144, in setup
      _install_setup_requires(attrs)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\__init__.py", line 139, in _install_setup_requires
      dist.fetch_build_eggs(dist.setup_requires)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\dist.py", line 716, in fetch_build_eggs
      resolved_dists = pkg_resources.working_set.resolve(
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\pkg_resources\__init__.py", line 780, in resolve
      dist = best[req.key] = env.best_match(
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\pkg_resources\__init__.py", line 1065, in best_match
      return self.obtain(req, installer)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\pkg_resources\__init__.py", line 1077, in obtain
      return installer(requirement)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\dist.py", line 786, in fetch_build_egg
      return cmd.easy_install(req)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\command\easy_install.py", line 679, in easy_install
      return self.install_item(spec, dist.location, tmpdir, deps)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\command\easy_install.py", line 705, in install_item
      dists = self.install_eggs(spec, download, tmpdir)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\command\easy_install.py", line 890, in install_eggs
      return self.build_and_install(setup_script, setup_base)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\command\easy_install.py", line 1158, in build_and_install
      self.run_setup(setup_script, setup_base, args)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\command\easy_install.py", line 1144, in run_setup
      run_setup(setup_script, args)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 253, in run_setup
      raise
    File "C:\Users\rlpiq\AppData\Local\Programs\Python\Python38-32\lib\contextlib.py", line 131, in __exit__
      self.gen.throw(type, value, traceback)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 195, in setup_context
      yield
    File "C:\Users\rlpiq\AppData\Local\Programs\Python\Python38-32\lib\contextlib.py", line 131, in __exit__
      self.gen.throw(type, value, traceback)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 166, in save_modules
      saved_exc.resume()
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 141, in resume
      six.reraise(type, exc, self._tb)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\_vendor\six.py", line 685, in reraise
      raise value.with_traceback(tb)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 154, in save_modules
      yield saved
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 195, in setup_context
      yield
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 250, in run_setup
      _execfile(setup_script, ns)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\sandbox.py", line 45, in _execfile
      exec(code, globals, locals)
    File "C:\Users\rlpiq\AppData\Local\Temp\easy_install-0wy4r5md\cffi-1.13.2\setup.py", line 127, in <module>
    File "C:\Users\rlpiq\AppData\Local\Temp\easy_install-0wy4r5md\cffi-1.13.2\setup.py", line 105, in uses_msvc
    File "C:\Users\rlpiq\AppData\Local\Programs\Python\Python38-32\lib\distutils\command\config.py", line 225, in try_compile
      self._compile(body, headers, include_dirs, lang)
    File "C:\Users\rlpiq\AppData\Local\Programs\Python\Python38-32\lib\distutils\command\config.py", line 132, in _compile
      self.compiler.compile([src], include_dirs=include_dirs)
    File "C:\Users\rlpiq\AppData\Local\Programs\Python\Python38-32\lib\distutils\_msvccompiler.py", line 360, in compile
      self.initialize()
    File "C:\Users\rlpiq\AppData\Local\Programs\Python\Python38-32\lib\distutils\_msvccompiler.py", line 253, in initialize
      vc_env = _get_vc_env(plat_spec)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\msvc.py", line 185, in msvc14_get_vc_env
      return EnvironmentInfo(plat_spec, vc_min_ver=14.0).return_env()
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\msvc.py", line 843, in __init__
      self.si = SystemInfo(self.ri, vc_ver)
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\msvc.py", line 485, in __init__
      self.vc_ver = vc_ver or self._find_latest_available_vc_ver()
    File "c:\users\rlpiq\freqtrade\.env\lib\site-packages\setuptools\msvc.py", line 492, in _find_latest_available_vc_ver
      raise distutils.errors.DistutilsPlatformError(err)
  distutils.errors.DistutilsPlatformError: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": https://visualstudio.microsoft.com/downloads/
  ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

(.env) C:\Users\rlpiq\freqtrade>
```

it works halfway then stops when its installing pycares? Im really lost and been working on this for a while. :(

**Comment by hroff-1902** [comment]
It says you explicitly:
> Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": https://visualstudio.microsoft.com/downloads/

The reason is that python 3.8 is pretty new and not all packages from dependencies are available for it as wheels (precompiled binary distributions), so the installation procedure tries to build them. So you should either install C/C++ build tools or change python installed at your system to python 3,6 or 3.7 and repeat installation from the beginning.

**Comment by xmatthias** [comment]
Also check this out in the [installation documentation for windows](https://www.freqtrade.io/en/latest/installation/#error-during-installation-under-windows) - which lists exactly this as a potential problem (solution is the same though ...) - since it's quite common on windows ...

**Comment by Theagainmen** [comment]
Note to windows users, you can install freqtrade using python 3.8 but you won't be able to install the hyperopt and plot functions. It does work on 3.7.https://app.slack.com/client/T6UE5AM9B/C6V6YH3N2/thread/C6V6YH3N2-1572095609.199100

**Comment by hroff-1902** [comment]
> but you won't be able to install the hyperopt and plot functions

More correct is: you won't be able to install the hyperopt and plot functions without resolving problems that appear during installation in this case.

**Comment by 007egleagent** [comment]
Hey, the issue got fixed! I had to download python 3.6 and then it worked. version 3.8 was not working

**Comment by xmatthias** [comment]
Great, good to know - thanks :+1:
